bgpd/bgpd.c

```diff
@@ -3875,6 +3875,32 @@ struct peer *peer_lookup_by_hostname(struct bgp *bgp, const char *hostname)
 	return NULL;
 }
 
+struct peer *peer_lookup_by_host(struct bgp *bgp, const char *host)
+{
+	struct peer *peer;
+	struct listnode *node, *nnode;
+
+	if (!host)
+		return NULL;
+
+	if (bgp != NULL) {
+		for (ALL_LIST_ELEMENTS(bgp->peer, node, nnode, peer))
+			if (peer->hostname && !strcmp(peer->host, host) &&
```

Comment:
I suggest using `strmatch()`.

Reply:
Copied from peer_lookup_by_hostname with some modifications. OK if you suggest.

Reply:
Fixed in #11522

Reply:
Topotest for RR case added in #11522